docs/coding-guidelines.md

```diff
@@ -429,9 +425,82 @@ for the following situations. Note that these changes should still be recorded a
   breaking change. For this reason, the deprecation process should only be skipped when it is not expected that
   the function is commonly passed as a value.
 
-### Configuration changes
+### End-user impacting changes
 
-See [docs/component-stability.md](component-stability.md) for more information on how to handle configuration changes.
+For end user breaking changes, we follow the [feature gate](https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate#feature-lifecycle)
+approach. This is a well-known approach in other projects such as Kubernetes. A feature gate has
+three stages: alpha, beta and stable. The intent of these stages is to decouple other software
+changes from the breaking change; some users may adopt the change early, while other users may delay
+its adoption.
+
+#### Feature gate IDs
+
+Feature gate IDs should be namespaced using dots to denote the hierarchy. The namespace should be as
+specific as possible; in particular, for feature gates specific to a certain component the ID should
+have the following structure: `<component kind>.<component type>.<base ID>`. The "base ID" should be
+written with a verb that describes what happens when the feature gate is enabled. For example, if
+you want to add a feature gate for the OTLP receiver that changes the default endpoint to bind to an
+unspecified host, you could name your feature gate `receiver.otlp.UseUnspecifiedHostAsDefaultHost`.
+
+#### Lifecycle of a breaking change
+
+##### Alpha stage
+
+At the alpha stage, the change is opt-in. At this stage we want to notify users that a change is
+coming, so they can start preparing for it and we have some early adopters that provide us with
+feedback. Consider the following items before the initial release of an alpha feature gate:
+
+* If **docs and examples** can be updated in a way that prevents the breaking change from affecting
+  users, this is the time to update them\!  
+* Provide users with tools to understand the breaking change  
+  * Create or update a **Github issue** to document what the change is about, who it affects and
```

Comment:
Should we make this optional? Alternatively, should we add a template to make it easier to create these issues?

Reply:
I made it optional, I think a template would make sense but I don't want to block this PR on that